### PR TITLE
feat(wallet): show solana networks

### PR DESCRIPTION
**Pull request by Pavneet-Sing**
- Solana auto creation
- Solana account and network link
- Show solana network on Portfio and dapps panel

Resolves https://github.com/brave/brave-browser/issues/23647
Resolves https://github.com/brave/brave-browser/issues/24011

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- Should be able to see Solana networks (localhost is not visible in release mode)
- Should see a pop-up to create Solana account for the first time
- Should see the solana as selected network on Portfolio and dapps panel
- Should be able to change networks and see the corresponding default account  (via panel)

https://user-images.githubusercontent.com/11755381/179816782-e1b313f3-0d16-4c35-9865-2c65dc73a435.mp4

### Additional test plan info:
- Portfolio or dapps panel screen does not show the updated solana balance, some refactoring will be a part of issue: https://github.com/brave/brave-browser/issues/23809
- Dapps panel is only used for testing, not a part of initial MVP

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/app/domain/CryptoModel.java [hunk 1]
```diff
@@ -235,14 +238,14 @@ public void processDecryptRequest(boolean isApproved, Origin origin) {
...
-                mContext.getString(R.string.brave_wallet_create_account_ethereum_description),
-                "Ethereum", CoinType.ETH, R.drawable.eth));
         if (isSolanaEnabled()) {
             cryptoAccountTypeInfos.add(new CryptoAccountTypeInfo(
                     mContext.getString(R.string.brave_wallet_create_account_solana_description),
                     "Solana", CoinType.SOL, R.drawable.ic_sol_asset_icon));
         }
+        cryptoAccountTypeInfos.add(new CryptoAccountTypeInfo(
+                mContext.getString(R.string.brave_wallet_create_account_ethereum_description),
+                "Ethereum", CoinType.ETH, R.drawable.eth));
```
Comment (qamarngr) [inline android/java/org/chromium/chrome/browser/app/domain/CryptoModel.java:248]: nit: "Ethereum" should be string res? same for Solana

**Review by qamarngr — APPROVED** [review]
LGTM

**Comment by qamarngr** [comment]
CI is failing for `noplatform` Please resolve that, maybe rebasing again will resolve

**Comment by Pavneet-Sing** [comment]
> CI is failing for `noplatform` Please resolve that, maybe rebasing again will resolve

yes, Already updated about this in slack, I will do it soon, thanks!

**Comment by srirambv** [comment]
Verification passed on Oppo Reno 5 with Android 12 running 1.43.39 x64 nightly build
- Verified steps from PR
- Verified able to create an account when Network dropdown is selected
- Verified selecting Solana for the first time shows popup to create the account
- Verified able to see Solana in the network drop down
- Verified able to select Solana from network drop down
- Encountered [#24193](https://github.com/brave/brave-browser/issues/24193) & [#24194](https://github.com/brave/brave-browser/issues/24193)

https://user-images.githubusercontent.com/17010094/180280227-4f29a27c-40b4-4990-936a-10abdb65241a.mp4

**Comment by Pavneet-Sing** [comment]
@srirambv both encountered issues are the same issue.

**Pull request merged**